stdlib/optparse.pyi

```diff
@@ -279,14 +279,11 @@ class OptionParser(OptionContainer):
     def get_prog_name(self) -> str: ...
     def get_usage(self) -> str: ...
     def get_version(self) -> str: ...
-    @overload
-    def parse_args(self, args: None = None, values: Values | None = None) -> tuple[Values, list[str]]: ...
-    @overload
-    def parse_args(self, args: Sequence[AnyStr], values: Values | None = None) -> tuple[Values, list[AnyStr]]: ...
```

Comment:
I want to explain the double change of `parse_args`'s `args` annotation. Consider this snippet:

```python
import optparse

parser = optparse.OptionParser()
parser.add_option("--target-dir", dest="target_dir", help="Target directory")
values, rest = parser.parse_args(["--target-dir", "/tmp"])
print(values, rest)
```
if you run it, it prints: `{'target_dir': '/tmp'} []`
1. `args` is now annotated as a list instead of a sequence. If you make this change to the above snippet:
   ```diff
   - values, rest = parser.parse_args(["--target-dir", "/tmp"])
   + values, rest = parser.parse_args(("--target-dir", "/tmp"))
   ```
   you get: `AttributeError: 'tuple' object has no attribute 'pop'`
2. `args` is now a list of strings, not `AnyStr`. If you make this change to the above snippet instead:
   ```diff
   - values, rest = parser.parse_args(["--target-dir", "/tmp"])
   + values, rest = parser.parse_args([b"--target-dir", b"/tmp"])
   ```
   you get: `{'target_dir': None} [b'--target-dir', b'/tmp']`
   This is not an error but it is clearly what you intended as the program failed to parse the arguments in this case.